GitTfsTest/TestHelpers/AssertTimeout.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Sep.Git.Tfs.Test.TestHelpers
+{
+    public static class AssertTimeout
+    {
+        /// <summary>
+        /// Fails tests which would otherwise run indefinitely.
+        /// </summary>
+        public static void For(TimeSpan limit, Action actionToLimit, string message = null)
+        {
+            if (actionToLimit == null) throw new ArgumentNullException("actionToLimit");
+
+            var thread = new Thread(actionToLimit.Invoke)
+            {
+                Name = "AssertTimeout " + actionToLimit.Method.Name
+            };
+
+            thread.Start();
+
+            if (Debugger.IsAttached)
+            {
+                thread.Join();
```

Comment:
@spraints Good catch! I overlooked that. Every threadstart should be wrapped in `try...catch` because otherwise instead of the caller being able to handle it, the entire process goes down.